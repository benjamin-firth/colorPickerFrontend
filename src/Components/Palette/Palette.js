import React from 'react';
import './Palette.scss';
import PropTypes from 'prop-types';

function Palette() {
  return (
    <section>
      <p>Palette 1</p>
      <div class='color1'></div>
      <div class='color2'></div>
      <div class='color3'></div>
      <div class='color4'></div>
      <div class='color5'></div>
      <button class='delete-palette-button'>X</button>
    </section>
  );
}

export default Palette;
