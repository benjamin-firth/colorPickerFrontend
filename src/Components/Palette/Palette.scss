.saved-palette-container {
  display: flex;
  align-items: center;
  margin-top: .5vw;
  button {
    background-color: transparent;
    border: none;
  };
  p {
    margin: 0;
    font-size: 1.8vw;
  }
};

.saved-palette-color {
  border: 1px solid black;
  border-radius: 200px 5px 200px 5px/5px 255px 5px 215px;
  margin: .2vw;
  width: 2.4vw;
  height: 2.4vw;
}
