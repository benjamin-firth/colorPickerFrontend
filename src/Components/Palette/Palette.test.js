import * as React from 'react';
import { render } from '@testing-library/react';
import Palette from './Palette';
import { shallow } from 'enzyme';
import { useDispatch } from 'react-redux';

jest.mock("react-redux", () => ({
  useSelector: jest.fn(fn => fn()),
  useDispatch: () => jest.fn()
}));

describe('Palette', () => {
  let wrapper;
  let mockPalette = {
    name: 'test',
    project_id: 2,
    color1: 'FFF',
    color2: 'FFF',
    color3: 'FFF',
    color4: 'FFF',
    color5: 'FFF',
    id: 3
  };

  beforeEach(() => {
    wrapper = shallow(<Palette palette={mockPalette}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
