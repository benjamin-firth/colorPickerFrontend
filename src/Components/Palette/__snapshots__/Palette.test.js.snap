// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match the snapshot 1`] = `
<section
  className="saved-palette-container"
>
  <p>
    test
    :
  </p>
  <div
    className="saved-palette-color"
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  />
  <div
    className="saved-palette-color"
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  />
  <div
    className="saved-palette-color"
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  />
  <div
    className="saved-palette-color"
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  />
  <div
    className="saved-palette-color"
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  />
  <button
    className="delete-palette-button"
    onClick={[Function]}
  >
    X
  </button>
</section>
`;
